scripts/react-compiler-compliance-check.ts

```diff
@@ -436,16 +498,152 @@ async function filterResultsByDiff(
+
+        if (!addedFileFailuresMap) {
+            return;
+        }
+
+        for (const [failureKey, failure] of addedFileFailuresMap) {
+            nonAutoMemoEnforcedFailures.set(failureKey, failure);
+        }
+    }
+
+    // Check all added files for manual memoization keywords and attach React compiler failures.
+    // If no manual memoization keywords are found add the failures back to the regular failures.
+    const addedComponentFailures: EnforcedAddedComponentFailureMap = new Map();
+    for (const addedFilePath of addedDiffFiles) {
+        const source = readSourceFile(addedFilePath);
+        if (!source || NO_MANUAL_MEMO_DIRECTIVE_PATTERN.test(source)) {
+            addNonAutoMemoEnforcedFailures(addedFilePath);
+            continue;
+        }
+
+        const manualMemoizationMatches = findManualMemoizationMatches(source);
+
+        if (manualMemoizationMatches.length === 0) {
+            addNonAutoMemoEnforcedFailures(addedFilePath);
+            continue;
+        }
+
+        const manualMemoFailure: ManualMemoFailure = {
+            manualMemoizationMatches,
+            compilerFailures: addedFileFailures.get(addedFilePath),
+        };
+        addedComponentFailures.set(addedFilePath, manualMemoFailure);
+    }
+
+    return {
+        nonAutoMemoEnforcedFailures,
+        addedComponentFailures,
+    };
+}
+
+/**
+ * Finds all manual memoization keywords matches in source file and returns their line and column numbers.
+ * @param source - The source code to search for manual memoization matches
+ * @returns An array of manual memoization matches
+ */
+function findManualMemoizationMatches(source: string): ManualMemoizationMatch[] {
+    const matches: ManualMemoizationMatch[] = [];
+
+    for (const keyword of Object.keys(MANUAL_MEMOIZATION_PATTERNS) as ManualMemoizationKeyword[]) {
+        const regex = MANUAL_MEMOIZATION_PATTERNS[keyword];
+        const regexMatches = source.matchAll(regex);
+
+        for (const regexMatch of regexMatches) {
+            const matchIndex = regexMatch.index;
+            if (matchIndex === undefined) {
+                continue;
+            }
+            const {line, column} = getLineAndColumnFromIndex(source, matchIndex);
+            matches.push({keyword, line, column});
+        }
+    }
+
+    // Sort matches by line number first, then by column number
+    matches.sort((a, b) => {
+        if (a.line !== b.line) {
+            return a.line - b.line;
+        }
+        return a.column - b.column;
+    });
+
+    return matches;
+}
+
+function getLineAndColumnFromIndex(source: string, index: number): {line: number; column: number} {
+    const substring = source.slice(0, index);
+    const line = substring.split('\n').length;
+    const lastLineBreakIndex = substring.lastIndexOf('\n');
+    const column = lastLineBreakIndex === -1 ? index + 1 : index - lastLineBreakIndex;
+    return {line, column};
+}
+
+function readSourceFile(filePath: string): string | null {
```

Comment:
NAB maybe this function and `getLineAndColumnFromIndex` could be put into a new `scripts/utils/FileUtils.ts`

Reply:
On a second look, we actually don't need this extra function so i refactored this part...